standard_revisions/match_variables_from_two_versions_of_a_dataset.py

```diff
@@ -0,0 +1,313 @@
+    unmatched_new = new_variables[~new_variables['name'].isin(mapping['name_new'])].reset_index(drop=True)
+    if len(unmatched_old) > 0:
+        print("\nUnmatched variables in the old dataset:")
+        for i, row in unmatched_old.iterrows():
+            print(f"  {row['name']} ({row['id']})")
+    else:
+        print("\nAll variables in the old dataset have been matched.")
+    if len(unmatched_new) > 0:
+        print("\nUnmatched variables in the new dataset:")
+        for i, row in unmatched_new.iterrows():
+            print(f"  {row['name']} ({row['id']})")
+    else:
+        print("\nAll variables in the new dataset have been matched.")
+
+
+def save_variable_replacements_file(mapping, output_file=OUTPUT_FILE):
+    """Save a json file with the mapping from old to new variable ids.
+
+    Parameters
+    ----------
+    mapping : pd.DataFrame
+        Mapping table from old variable name and id to new variable name and id.
+    output_file : str
+        Path to output file.
+
+    """
+    # Create a dictionary mapping from old variable id to new variable id.
+    mapping_indexes = mapping[['id_old', 'id_new']].set_index('id_old').to_dict()['id_new']
+    mapping_indexes = {str(key): str(mapping_indexes[key]) for key in mapping_indexes}
+
+    print(f"Saving index mapping to json file: {output_file}")
+    save_data_to_json_file(data=mapping_indexes, json_file=output_file, **{'indent': 4, 'sort_keys': True})
+
+
+def main(old_dataset_name, new_dataset_name, omit_identical=OMIT_IDENTICAL, matching_function=MATCHING_FUNCTION,
+         output_file=OUTPUT_FILE):
+    with get_connection() as db_conn:
+        # Get old and new dataset ids.
+        old_dataset_id = get_dataset_id(db_conn=db_conn, dataset_name=old_dataset_name)
+        new_dataset_id = get_dataset_id(db_conn=db_conn, dataset_name=new_dataset_name)
+
+        # Get variables for old and new datasets.
+        old_variables = get_variables_in_dataset(db_conn=db_conn, dataset_id=old_dataset_id)
+        new_variables = get_variables_in_dataset(db_conn=db_conn, dataset_id=new_dataset_id)
+    
+    # Manually map old variable names to new variable names.
+    mapping = map_old_and_new_variables(
+        old_variables=old_variables, new_variables=new_variables, omit_identical=omit_identical,
+        matching_function=matching_function)
+
+    # Display summary.
+    display_summary(old_variables=old_variables, new_variables=new_variables, mapping=mapping)
+
+    # Save mapping to json file.
+    save_variable_replacements_file(mapping, output_file=output_file)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description="Map variable names from an old version of a dataset to variables of a new dataset."
+    )
+    parser.add_argument(
+        "-f",
+        "--output_file",
+        default=OUTPUT_FILE,
+        help=f"Path to output json file. Default: "
+        f"{OUTPUT_FILE}",
+    )
+    parser.add_argument(
```

Comment:
There are cases where the new dataset is replacing multiple old datasets, so ideally this argument would be `--old_dataset_names` that would accept an array of old dataset names. 

Though this seems like a low priority fix. A work-around for now is that I could just run this script once for each old dataset, with a different `--output_file` arg each time and then merge the `variable_replacements.json` files afterwards.

e.g. for World Bank WDI, the dataset was previously broken into a number of smaller datasets. So the first bulk import I ran, I had to match old variables from multiple datasets to new variables in the new dataset I was uploading.

Reply:
Let's leave this for a future improvement.